store/src/java/com/zimbra/cs/security/sasl/ZimbraAuthenticator.java

```diff
@@ -92,24 +93,29 @@ public ZimbraAuthenticator(AuthenticatorUser user) {
         }
 
         // make sure that the authentication account is valid
-        Account authAccount = prov.get(Key.AccountBy.name, authenticateId, at);
+        Account authAccount = prov.get(at.isAdmin() ? Key.AccountBy.adminName : Key.AccountBy.name, authenticateId, at);
         if (authAccount == null)
             return null;
 
         // make sure the auth token belongs to authenticatedId
         if (!at.getAccountId().equalsIgnoreCase(authAccount.getId()))
             return null;
-        
+
         // make sure the protocol is enabled for the user
         if (!isProtocolEnabled(authAccount, protocol)) {
             ZimbraLog.account.info("Authentication failed - %s not enabled for %s", protocol, authAccount.getName());
             return null;
         }
 
         // if necessary, check that the authenticated user can authorize as the target user
-        Account targetAcct = authorize(authAccount, username, AuthToken.isAnyAdmin(at));
-        if (targetAcct != null)
+        if (at.isAdmin() && username.equals(authenticateId) && username.equals(LC.zimbra_ldap_user.value())) {
```

Comment:
Yeah, in fact this made me realize that this bypassing of the authorize() method means that other admin accounts won't be able to use this command.